File: stepik-java-api/src/main/java/org/stepik/api/objects/AbstractObjectWithStringId.java
package org.stepik.api.objects;

import org.jetbrains.annotations.Nullable;

/**
 * @author meanmail
 */
public class AbstractObjectWithStringId {
    private String id;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        AbstractObjectWithStringId that = (AbstractObjectWithStringId) o;

        return id != null ? id.equals(that.id) : that.id == null;
    }

    @Override
    public int hashCode() {
        return id != null ? id.hashCode() : 0;
    }

    @Nullable
    public String getId() {
        return id;
    }

    public void setId(@Nullable String id) {
        this.id = id;
    }
}

File: stepik-java-api/src/main/java/org/stepik/api/objects/AbstractObjectWithStringId.kt
package org.stepik.api.objects

import com.google.gson.annotations.JsonAdapter
import org.stepik.api.client.serialization.DefaultAsEmptyStringAdapter


open class AbstractObjectWithStringId {
    @JsonAdapter(DefaultAsEmptyStringAdapter::class, nullSafe = false)
    var id: String = ""

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as AbstractObjectWithStringId

        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        return id.hashCode()
    }


}
